presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -72,7 +72,7 @@ private static SymbolStatsEstimate subtractColumnStats(
         double nullsCountRight = rightStats.getNullsFraction() * rightRowCount;
         double totalSizeLeft = (leftRowCount - nullsCountLeft) * leftStats.getAverageRowSize();
         double totalSizeRight = (rightRowCount - nullsCountRight) * rightStats.getAverageRowSize();
-        double newNullsFraction = (nullsCountLeft - nullsCountRight) / newRowCount;
+        double newNullsFraction = Math.max(nullsCountLeft - nullsCountRight, 0.0) / newRowCount;
```

Comment:
Were you able to reproduce this somehow?

Reply:
`checkArgument(nullsCountLeft >= nullsCountRight)`.

You cannot be sure that it will happen as this is kind of multiset substraction.